fem/gridfunc.hpp

```diff
@@ -586,6 +586,10 @@ class GridFunction : public Vector
       return ComputeLpError(infinity(), exsol, NULL, NULL, irs);
    }
 
+   virtual double ComputeL1Error(Coefficient *exsol[],
```

Comment:
This is following the pattern for the passing of `ComputeL2Error`, where there's an alternative method for this array of pointers of coefficients. The `ComputeW11Error` method is used in the pgrid function version, and what I noticed is that because that function isn't marked as override, there doesn't exist a matching `gridfunc.hpp` original version of the method (this is also the textbook example of why `override` should be used). Matching the usage on the array of pointers `ComputeL2Error` call in `pgridfunc.hpp`, I add the serial call in grid function which would be overriden.

Reply:
Thanks for the explanation. I see now that this new function adds what was missing to follow the pattern used by L2. I wasn't aware that these norms were computed differently, for slightly different function signatures, and I am curious about the reasoning behind that. After briefly reviewing the existing functions `ComputeW11Error` and `ComputeLpError` for the L1 case, it appears that they do the same computation with different implementation (e.g. one uses GetValues whereas the other computes values via shape functions). I wonder whether they are really the same up to round-off, and if so, why do we use different functions that compute the same thing? Maybe someone familiar with the history of these functions can comment. Hopefully some clarification can lead to more documentation, since I am a little confused still. It would be nice to have documentation on these functions to say what they compute and whether they are the same as other functions with different signatures.